src/test/java/kitchenpos/application/MenuServiceTest.java

```diff
@@ -0,0 +1,189 @@
+            //given
+            menu.setPrice(BigDecimal.valueOf(-1));
+
+            //when & then
+            assertThatThrownBy(() -> menuService.create(menu))
+                    .isInstanceOf(IllegalArgumentException.class);
+        }
+
+        @Test
+        @DisplayName("메뉴가 포함될 메뉴그룹이 존재하지 않으면, 예외를 던진다.")
+        void fail_noExistMenuGroup() {
+            //given
+            given(menuGroupDao.existsById(menu.getMenuGroupId()))
+                    .willReturn(false);
+
+            //when & then
+            assertThatThrownBy(() -> menuService.create(menu))
+                    .isInstanceOf(IllegalArgumentException.class);
+        }
+
+        @Test
+        @DisplayName("메뉴에 포함될 상품들이 존재하지 않으면, 예외를 던진다.")
+        void fail_noExistMenuProduct() {
+            //given
+            given(productDao.findById(PRODUCT_A_ID))
+                    .willReturn(Optional.empty());
+
+            //when & then
+            assertThatThrownBy(() -> menuService.create(menu))
+                    .isInstanceOf(IllegalArgumentException.class);
+        }
+
+        @Test
+        @DisplayName("메뉴의 가격보다 포함된 상품들의 총합이 싸면, 예외를 던진다.")
+        void fail_sumIsCheaperThenPrice() {
+            //given
+            productA.setPrice(BigDecimal.valueOf(100));
+            menuProductA.setQuantity(1);
+            productB.setPrice(BigDecimal.valueOf(100));
+            menuProductB.setQuantity(1);
+
+            menu.setPrice(BigDecimal.valueOf(201));
+
+            //when & then
+            assertThatThrownBy(() -> menuService.create(menu))
+                    .isInstanceOf(IllegalArgumentException.class);
+        }
+    }
+
+    @Nested
+    @DisplayName("list 메서드는")
+    class ListMethod {
+        private static final long PRODUCT_A_ID = 1L;
+        private static final long PRODUCT_B_ID = 2L;
+        private static final long MENU_ID = 1L;
+        private static final long MENU_GROUP_ID = 1L;
+        private static final long MENU_PRICE = 10000;
+
+        private MenuProduct menuProductA;
+        private MenuProduct menuProductB;
+        private Menu menu;
+
+        @Test
+        @DisplayName("전체 메뉴를 조회할 때, 메뉴상품도 같이 조회할 수 있다.")
+        void success_getMenuProducts() {
+            menuProductA = ClassConstructor.menuProduct(null, MENU_ID, PRODUCT_A_ID, 2);
+            menuProductB = ClassConstructor.menuProduct(null, MENU_ID, PRODUCT_B_ID, 2);
+
+            menu = ClassConstructor.menu(MENU_ID, "메뉴 이름", BigDecimal.valueOf(MENU_PRICE), MENU_GROUP_ID,
+                    Arrays.asList(menuProductA, menuProductB));
+
+            given(menuProductDao.findAllByMenuId(1L))
+                    .willReturn(Arrays.asList(menuProductA, menuProductB));
+
+            //when
+            List<Menu> actual = menuService.list();
+            Menu actualMenu = actual.stream()
+                    .filter(it -> it.getId().equals(1L))
+                    .findFirst()
+                    .orElseThrow(IllegalArgumentException::new);
```

Comment:
```suggestion
            Menu actualMenu = actual.iterator().next();
```

1번 메뉴에 대해서만 검증목적이라면 이렇게 해보는 건 어떠세요? 😀
단순 의견입니다 ㅎㅎ

Reply:
이부분이 무지 불만이었는데 시간이 모자라서 그냥 넘겼었는데 그걸 캐치해주시네요 역시 리차드👍🏻
이렇게 풀 수 있군요!! 너무 좋죠ㅋㅋㅋㅋㅋ
바로 반영했습니다!